### PR TITLE
Refactor bundle_test.go fake registry

### PR DESCRIPTION
**Pull request by DennisDenuto**
- Bumps google-container-registry library
- Uses a [v2 compliant registry](https://github.com/vmware-tanzu/carvel-imgpkg/blob/508de4948bdd754ae0b0caca286e7dc58f1f63b0/vendor/github.com/google/go-containerregistry/pkg/registry/registry.go#L24) for tests to read/write to. This is a change from having to 'mock' out the `ImagesMetadata` functions.
  - This pattern can be (but hasn't in this PR) apply to [helpers.fake_registry](https://github.com/vmware-tanzu/carvel-imgpkg/blob/56c87b10541d207cf38fad9984e49caba78d4f91/test/helpers/fake_registry.go#L27-L26)
  - Should make understanding our test code simpler (since there is less logic around mocking our registry object)

### CODE REVIEW

File: pkg/imgpkg/bundle/bundle_test.go [hunk 1]
```diff
@@ -82,7 +84,12 @@ func TestPullAllNestedBundlesWritingContentsToDisk(t *testing.T) {
 		defer fakeImagesMetadataBuilder.CleanUp()
 
 		fakeImagesMetadataBuilder.WithBundleFromPath("repo/some-bundle-name", "test_assets/bundle").WithEveryImageFrom("test_assets/image_with_config", map[string]string{})
-		subject := bundle.NewBundle("repo/some-bundle-name", fakeImagesMetadataBuilder.Build())
+
+		build := fakeImagesMetadataBuilder.Build()
+		u, err := url.Parse(fakeImagesMetadataBuilder.server.URL)
```
Comment (joaopapereira) [inline pkg/imgpkg/bundle/bundle_test.go]: Isn't this equivalent to
`fakeImagesMetadataBuilder.relocateToTestServer("repo/some-bundle-name")`

File: pkg/imgpkg/bundle/fake_images_metadata_builder_test.go [hunk 2]
```diff
@@ -127,41 +86,37 @@ func (r *FakeImagesMetadataBuilder) WithImageFromPath(imageNameFromTest string,
...
+func (r *FakeTestRegistryBuilder) CleanUp() {
 	for _, tarPath := range r.state {
 		os.Remove(filepath.Join(tarPath.path, ".imgpkg", "images.yml"))
 	}
+	if r.server != nil {
+		r.server.Close()
+	}
+}
+
+func (r *FakeTestRegistryBuilder) relocateToTestServer(repo string) string {
```
Comment (joaopapereira) [inline pkg/imgpkg/bundle/fake_images_metadata_builder_test.go]:
```suggestion
func (r *FakeTestRegistryBuilder) referenceOnTestServer(repo string) string {
```

Since there are no relocations in `imgpkg` maybe it makes sense to call relocation to this method. The above is a suggestion.

File: pkg/imgpkg/bundle/fake_images_metadata_builder_test.go [hunk 3]
```diff
@@ -8,94 +8,53 @@ import (
...
+	"github.com/k14s/imgpkg/pkg/imgpkg/imagelayers"
 	"github.com/k14s/imgpkg/pkg/imgpkg/lockconfig"
+	"github.com/stretchr/testify/assert"
 )
 
-type FakeImagesMetadataBuilder struct {
-	state map[string]*ImageWithTarPath
-	t     *testing.T
+type FakeTestRegistryBuilder struct {
+	state  map[string]*ImageWithTarPath
```
Comment (joaopapereira) [inline pkg/imgpkg/bundle/fake_images_metadata_builder_test.go]:
```suggestion
	images  map[string]*ImageWithTarPath
```

This is no longer the state but the list of images we are going to push to the registry.

File: pkg/imgpkg/bundle/fake_images_metadata_builder_test.go (hunk 2 above)
Comment (joaopapereira) [reply pkg/imgpkg/bundle/fake_images_metadata_builder_test.go]: Looking at the rest of the functions, is there any reason for this function to be private?

**Review by joaopapereira — CHANGES_REQUESTED** [review]
Nice to see we are keeping our dependencies up to date 😄. It looks like a great improvement since the implementation of our fake was starting to get pretty hard to follow.

Since we are simplifying this fake maybe it would make sense to do the same for the `helpers` package version. 
The major difference that I see is that the other fake represents multiple registries with different URLs, which might become challenging. Unless we spin up multiple registries via `NewFakeImagesMetadataBuilder`

File: pkg/imgpkg/bundle/bundle_test.go (hunk 1 above)
Comment (DennisDenuto) [reply pkg/imgpkg/bundle/bundle_test.go]: Good catch. Thx I made the change

File: pkg/imgpkg/bundle/fake_images_metadata_builder_test.go (hunk 2 above)
Comment (DennisDenuto) [reply pkg/imgpkg/bundle/fake_images_metadata_builder_test.go]: Good catch. Thx I made the change

File: pkg/imgpkg/bundle/fake_images_metadata_builder_test.go (hunk 3 above)
Comment (DennisDenuto) [reply pkg/imgpkg/bundle/fake_images_metadata_builder_test.go]: Good catch. Thx I made the change

**Review by joaopapereira — APPROVED** [review]
LGTM

**Comment by joaopapereira** [comment]
I talked with @DennisDenuto talked about the update of `helpers` package registry and agreed that we could do this in the next story that will touch the tests that will use this Fake.
